examples/systemcalls/systemcalls.c

```diff
@@ -74,24 +138,72 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     va_list args;
     va_start(args, count);
     char * command[count+1];
+    int status;
     int i;
+    int ret_dup;
+    pid_t process_id;
+
+    int fd = creat(outputfile, 0644);
+
+    printf("\n\ndo_exec_redirect: ************************************************");
+    printf("\nProcess id of current process in %d", getpid());
+
     for(i=0; i<count; i++)
     {
         command[i] = va_arg(args, char *);
+        //printf("\ncommand[%d] :%s", i, command[i]);
     }
+
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
 
+    /*******************fork********************/
+    process_id = fork();
+    if(process_id == -1)
+    {
+        printf("\nfork failed");
+        return false;
+    }
+    
 
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *   
-*/
+    /*******************Execv********************/
+    else if (process_id == 0) {
+        printf("\nExecutiing execv : ");
+        for(int i=0; i<count; i++)
+        {
+            printf("%s ", command[i]);
+        } 
+        printf("\n");
+
+        ret_dup = dup2(fd, STDOUT_FILENO);
```

Comment:
@rajatchaple think about how you could avoid duplicating logic with the function above.

Reply:
One way I can think about is using  > filepath

In here I will have to replace $ sign with \$ in a string, so that it does not expand $Home path variable to my absolute path.